kymatio/scattering3d/backend/torch_backend.py

```diff
@@ -58,11 +58,129 @@ def compute_integrals(input_array, integral_powers):
+            
+            Attributes
+            ----------
+            padding_module : ReflectionPad2d
+                Pads the input tensor using the reflection of the input
+                boundary.
+
+        """
+        pad_size_tmp = list(self.pad_size)
+
+        # This handles the case where the padding is equal to the image size
+        if pad_size_tmp[0] == self.input_size[0]:
+            pad_size_tmp[0] -= 1
+            pad_size_tmp[1] -= 1
+        if pad_size_tmp[2] == self.input_size[1]:
+            pad_size_tmp[2] -= 1
+            pad_size_tmp[3] -= 1
+        if pad_size_tmp[4] == self.input_size[2]:
+            pad_size_tmp[4] -= 1
+            pad_size_tmp[5] -= 1
+
+        self.pad_size_tmp = pad_size_tmp
+
+    def __call__(self, x):
+        """Applies padding and maps to complex.
+
+            Parameters
+            ----------
+            x : tensor
+                Real tensor input to be padded and sent to complex domain.
+            Returns
+            -------
+            output : tensor
+                Complex torch tensor that has been padded.
+
+        """
+        if not self.pre_pad:
+            # Pytorch expects its padding as [left, right, top, bottom]
+            pad_size_tmp = self.pad_size_tmp
+            x = pad(x, (pad_size_tmp[2], pad_size_tmp[3], 
+                pad_size_tmp[0], pad_size_tmp[1], pad_size_tmp[4],
+                pad_size_tmp[5]), mode='constant')
+
+            # Note: PyTorch is not effective to pad signals of size N-1 with N
+            # elements, thus we had to add this fix.
+            if self.pad_size[0] == self.input_size[0]:
+                x = torch.cat([x[:, 1, :, :, :, :].unsqueeze(2), x,
+                    x[:, x.shape[1] - 2, :, :, :, :].unsqueeze(2)], 2)
+            if self.pad_size[2] == self.input_size[1]:
+                x = torch.cat([x[:, :, 1, :, :, :].unsqueeze(3), x, 
+                    x[:, :, :, x.shape[3] - 2, :, :].unsqueeze(3)], 3)
+            if self.pad_size[4] == self.input_size[2]:
+                x = torch.cat([x[:, :, :, :, 1, :].unsqueeze(4), x, 
+                    x[:, :, :, :, x.shape[4] - 2, :].unsqueeze(4)], 4)
+
+            return x.unsqueeze(-1)
+
+def unpad(in_):
+    """Unpads input.
+
+        Slices the input tensor at indices between 1:-1.
+
+        Parameters
+        ----------
+        in_ : tensor
+            Input tensor.
+
+        Returns
+        -------
+        in_[..., 1:-1, 1:-1, 1:-1, :] : tensor
+            Output tensor. Unpadded input.
+
+    """
+    return in_[..., 1:-1, 1:-1, 1:-1]
+
+class SubsampleFourier(object):
+    def __call__(self, x, k):
+        complex_check(x)
+        contiguous_check(x)
+
+        y = x.view(-1,
+                    k, x.shape[1] // k,
+                    k, x.shape[2] // k,
+                    k, x.shape[3] // k,
+                    2)
+        out = y.mean(5, keepdim=False).mean(3, keepdim=False).mean(1, keepdim=False)
```

Comment:
i guess it is still not possible to take the mean over multiple axes yet?

Reply:
actually, it appears to be.